src/components/modal/modal.tsx

```diff
@@ -0,0 +1,116 @@
+    const mainClasses = clsx(
+        'w-full justify-between bg-white px-l pb-l',
+        ContentVariant.SETTINGS === 'settings' && 'w-full flex-row gap-l',
+        ContentVariant.FILE_UPLOAD === 'file_upload' && 'flex flex-col gap-s pb-s pt-l',
+    );
+    const footerClasses =
+        'flex w-full flex-row justify-between gap-l pt-xs p-l rounded-bl-s rounded-br-s bg-white';
+    const buttonClasses = 'hover:cursor-pointer basis-1/2';
+
+    return (
+        <Transition appear show={isOpen} as={Fragment}>
+            <Dialog initialFocus={initialFocusInputRef} onClose={onActionSecondary}>
+                <Transition.Child
+                    as={Fragment}
+                    enter="ease-out duration-300"
+                    enterFrom="opacity-0"
+                    enterTo="opacity-40"
+                    leave="ease-in duration-200"
+                    leaveFrom="opacity-100"
+                    leaveTo="opacity-0"
+                >
+                    <div
+                        role="backgrop"
+                        className="fixed inset-0 bg-primary-600"
+                        aria-hidden="true"
+                    />
+                </Transition.Child>
+
+                <div className="p-4 fixed inset-0 flex w-screen items-center justify-center">
+                    <div className={modalWidth}>
+                        <Transition.Child
+                            as={Fragment}
+                            enter="ease-out duration-300"
+                            enterFrom="opacity-0 scale-95"
+                            enterTo="opacity-100 scale-100"
+                            leave="ease-in duration-200"
+                            leaveFrom="opacity-100 scale-100"
+                            leaveTo="opacity-0 scale-95"
+                        >
+                            <Dialog.Panel className={dialogPanelClasses}>
+                                {title && (
+                                    <header className={headerClasses}>
+                                        <Heading size={HeadingSize.H3} className="grow text-white">
+                                            {title}
+                                        </Heading>
+                                        <button
+                                            name="Cancel"
+                                            className="fill-white hover:cursor-pointer"
+                                            onClick={onActionSecondary}
+                                        >
+                                            <IconCancel size={IconSize.M} />
+                                        </button>
+                                    </header>
+                                )}
+
+                                {contentVariant === 'settings' ? (
+                                    <main className={mainClasses}>{children}</main>
+                                ) : contentVariant === 'file_upload' ? (
```

Comment:
Since the content here is the same, is this check even needed? I guess you can delete this enum check and just add `<main className={mainClasses}>{children}</main>`